tests/auto-alex-v3.test.ts

```diff
@@ -0,0 +1,127 @@
+            tx.callPublicFn('auto-alex-v3-endpoint', 'request-redeem', [Cl.uint(dx)], wallet_3),
+            tx.callPublicFn('auto-alex-v3-endpoint', 'request-redeem', [Cl.uint(dx)], wallet_4),
+            // test revoke redeem request
+            tx.callPublicFn('auto-alex-v3-endpoint', 'revoke-redeem', [Cl.uint(4)], wallet_4),
+        ]);
+        response.map((e: any) => expect(e.result).toHaveClarityType(ClarityType.ResponseOk));
+
+        // console.log(`current block: ${simnet.blockHeight}`);
+        // console.log(`current cycle: ${simnet.callReadOnlyFn("SP102V8P0F7JX67ARQ77WEA3D3CFB5XW39REDT0AM.alex-staking-v2", "get-reward-cycle", [Cl.uint(simnet.blockHeight)], wallet_1).result.value.value}`);
+
+        const redeem_cycle = end_cycle + 2;
+
+        for (let cycle = 2; cycle < redeem_cycle; cycle++) {
+            simnet.mineEmptyBlocks(ACTIVATION_BLOCK + (cycle + 1) * 525 - simnet.blockHeight);
+            // console.log(`current block: ${simnet.blockHeight}`);
+            // console.log(`current cycle: ${simnet.callReadOnlyFn("SP102V8P0F7JX67ARQ77WEA3D3CFB5XW39REDT0AM.alex-staking-v2", "get-reward-cycle", [Cl.uint(simnet.blockHeight)], wallet_1).result.value.value}`);
+            // console.log(`alex-vault bal: ${simnet.callReadOnlyFn("SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.age000-governance-token", "get-balance-fixed", [Cl.principal("SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.alex-vault")], wallet_1).result.value.value}`);
+            // console.log(`auto-alex-v2 bal: ${simnet.callReadOnlyFn("SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.age000-governance-token", "get-balance-fixed", [Cl.principal("SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.auto-alex-v2")], wallet_1).result.value.value}`);
+            // console.log(`redeem: ${simnet.callReadOnlyFn('auto-alex-v3-endpoint', 'get-redeem-request-or-fail', [Cl.uint(2)], wallet_1).result.value.data.amount.value}`);
+            response = simnet.mineBlock([
+                tx.callPublicFn(contracts.oldAlex, 'mint-fixed', [Cl.uint(3), Cl.principal(contracts.autoAlexV2)], simnet.deployer),
+                tx.callPublicFn('auto-alex-v3-endpoint', 'rebase', [], simnet.deployer),
+                // you cannot finalize redeem until end cycle
+                tx.callPublicFn('auto-alex-v3-endpoint', 'finalize-redeem', [Cl.uint(2)], wallet_2),
+            ]);
+
+            // console.log(response[0].events);
+            expect(response[0].result).toHaveClarityType(ClarityType.ResponseOk);
+            expect(response[1].result).toHaveClarityType(ClarityType.ResponseOk);
+            expect(response[2].result).toBeErr(Cl.uint(10017));
+        }
+
+        simnet.mineEmptyBlocks(ACTIVATION_BLOCK + (redeem_cycle + 1) * 525 - simnet.blockHeight);
+
+        console.log(simnet.callReadOnlyFn(contracts.oldAlex, 'get-balance-fixed', [Cl.principal(simnet.deployer + '.auto-alex-v3')], wallet_1));
```

Comment:
What are the expected values here?

Reply:
It's just verbose output.
https://github.com/lisalab-io/liquid-stacking/pull/56/commits/20107362b60f7890c0538305e7089746f26b6ac3